src/Native/Native.proj

```diff
@@ -55,11 +55,16 @@
           DependsOnTargets="GenerateNativeVersionFile">
     <PropertyGroup>
       <StripArgs Condition="'$(StripNativeSymbols)' == 'True'">--stripSymbols</StripArgs>
+      <BuildArgs>--configuration $(Configuration) --arch $(TargetArchitecture) $(StripArgs)</BuildArgs>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="!$(TargetArchitecture.StartsWith('arm'))">
+      <BuildArgs>$(BuildArgs) --mkllibpath $(NuGetPackageRoot)mlnetmkldeps/$(MlNetMklDepsVersion)/runtimes/$(PackageRid)/native</BuildArgs>
       <!-- mkllibrpath is used as rpath on macOS by binaries such as SymSDGNative and MklProxyNative to locate mkl library.
            We are shipping 2 native assemblies in different nuget packages, and one has a reference on the other and MacOS needs to have
            the rpath in the assembly to where it should load the referenced assembly and since .NET Core can run assemblies out of a NuGet cache,
            we need to add the NuGet cache relative location. -->
-      <BuildArgs>--configuration $(Configuration) --arch $(TargetArchitecture) $(StripArgs) --mkllibpath $(NuGetPackageRoot)mlnetmkldeps/$(MlNetMklDepsVersion)/runtimes/$(PackageRid)/native --mkllibrpath ../../../../../microsoft.ml.mkl.redist/$(Version)/runtimes/$(PackageRid)/native</BuildArgs>
+      <BuildArgs Condition="$([MSBuild]::IsOSPlatform('osx'))">$(BuildArgs) --mkllibrpath ../../../../../microsoft.ml.mkl.redist/$(Version)/runtimes/$(PackageRid)/native</BuildArgs>
```

Comment:
>../../../../../microsoft. [](http://example.com/codeflow?start=90&length=25)

should we change this to use the same we did with `mkllibpath`

Reply:
I looked into that relative path, it's actually burned into the final binary to support the scenario where someone is running from the nuget cache.  We need it to remain relative because absolute paths will not be correct on the user's machine.  That's what that comment above is trying to communicate.

Reply:
One thing that surprised me was that the host wasn't setting up the OS probing paths to handle this scenario, but I remember discussions long ago with Morgan about why it's difficult to do that.  IIRC there isn't a universal way to tell all operating systems to probe for all load-libraries in a set of directories.  We can do that for DLLImport's but not for native->native loads.  Maybe @vitek-karas or @AaronRobinsonMSFT can shed some light on wether that's still the case.

Reply:
Unfortunately there isn't a good way. Windows has the `PATH` env and Linux has `LD_LIBRARY_PATH` so the host could set them early for library binding, but macOS is very tricky and doesn't permit setting the equivalent in all cases. Relative paths are still the most appropriate from what I recall.